### PR TITLE
Consolidate L2Config and L2ExecutionConfig

### PR DESCRIPTION
**Pull request by cygaar**
We now have an L2Config class so it makes sense to consolidate all L2 related configs in one class. This also makes the `deploy_l2` function more uniform as all functions can use the `L2Config`.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -79,6 +82,30 @@ def __init__(self):
...
+        # ==========================================================================================
+        # Geth Configuration
+
+        self.data_dir = "db/L2-execution"
+        """Geth data directory for op-geth node."""
+
+        self.chaindata_dir = f"{self.data_dir}/geth/chaindata"
+        """Directory storing chain data."""
+
+        self.chain_id = 42069
```
Comment (cygaar) [inline config.py:94]: I set this to be the chain_id in the getting-started docs, lmk if there should be a different default

Comment (norswap) [reply config.py:94]: I think down the line we want some parameters to be mandatory and this should probably be one of them. For now this is fine!

File: config.py [hunk 2]
```diff
@@ -79,6 +82,30 @@ def __init__(self):
...
+
+        self.verbosity = os.environ.get("GETH_VERBOSITY", 3)
+        """Geth verbosity level (from 0 to 5, see geth --help)."""
+
+        self.rpc_port = os.environ.get("RPC_PORT", 9545)
+        """Port to use for the http-based JSON-RPC server."""
+
+        self.ws_port = os.environ.get("WS_PORT", 9546)
+        """Port to use for the WebSocket-based JSON_RPC server."""
+
```
Comment (norswap) [inline config.py:108]: We probably should get rid of the env override. We don't use them and we haven't created them for the other config values.

File: config.py [hunk 3]
```diff
@@ -419,6 +446,11 @@ def use_devnet_config(self, paths: OPPaths):
 
         self.jwt_secret_path = paths.jwt_test_secret_path
 
+        # === Geth ===
```
Comment (norswap) [inline config.py:449]: Same comment as above.

File: config.py [hunk 4]
```diff
@@ -442,3 +474,8 @@ def use_devnet_config(self, paths: OPPaths):
         self.sub_safety_margin = 4
 
 ####################################################################################################
+
+L2_EXECUTION_DATA_DIR = "db/L2-execution"
+"""Directory to store the op-geth blockchain data."""
```
Comment (norswap) [inline config.py:479]: Ultimately this should be inlined into the `__init__` directly, but I supposed this still exists for the sake of the `clean` function?

**Review by norswap — APPROVED** [review]
Nice work! made a few comments. Might merge tomorrow anyway as I'll be working on the code and leaving the PR open will make it stale.

File: config.py (hunk 4 above)
Comment (cygaar) [reply config.py:479]: Yep, it's for the `clean` function? I replaced the call in `start_l2_execution_node` with `config.data_dir`.

**Pull request merged**